L1Trigger/L1TMuonOverlapPhase2/src/PtAssignmentNNRegression.cc

```diff
@@ -0,0 +1,308 @@
+      << " " << __FUNCTION__ << ":" << __LINE__ << " networkFile " << networkFile << std::endl;
+
+  lutNetworkFP->load(networkFile);
+
+  edm::LogImportant("OMTFReconstruction") << " " << __FUNCTION__ << ":" << __LINE__ << std::endl;
+}
+
+struct OmtfHit {
+  union {
+    unsigned long rawData = 0;
+
+    struct {
+      char layer;
+      char quality;
+      char z;
+      char valid;
+      short eta;
+      short phiDist;
+    };
+  };
+
+  OmtfHit(unsigned long rawData) : rawData(rawData) {}
+};
+
+bool omtfHitToEventInput(OmtfHit& hit, std::vector<float>& inputs, unsigned int omtfRefLayer, bool print) {
+  float offset = (omtfRefLayer << 7) + 64;
+
+  if (hit.valid) {
+    if ((hit.layer == 1 || hit.layer == 3 || hit.layer == 5) && hit.quality < 4)  ///TODO <<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+      return false;
+
+    int rangeFactor = 2;  //rangeFactor scales the hit.phiDist such that the event->inputs is smaller then 63
+    if (hit.layer == 1) {
+      rangeFactor = 8;
+    }
+    /*else if(hit.layer == 8 || hit.layer == 17) {
+            rangeFactor = 4;
+        }*/
+    else if (hit.layer == 3) {
+      rangeFactor = 4;
+    } else if (hit.layer == 9) {
+      rangeFactor = 1;
+    }
+    /*else {
+            rangeFactor = 2;
+        }
+		 */
+
+    rangeFactor *= 2;  //TODO !!!!!!!!!!!!!!!!!!!
+
+    if (std::abs(hit.phiDist) >= (63 * rangeFactor)) {
+      edm::LogImportant("OMTFReconstruction")  //<<" muonPt "<<omtfEvent.muonPt<<" omtfPt "<<omtfEvent.omtfPt
+          << " RefLayer " << omtfRefLayer << " layer " << int(hit.layer) << " hit.phiDist " << hit.phiDist << " valid "
+          << ((short)hit.valid) << " !!!!!!!!!!!!!!!!!!!!!!!!" << endl;
+      hit.phiDist = copysign(63 * rangeFactor, hit.phiDist);
+    }
+
+    inputs.at(hit.layer) = (float)hit.phiDist / (float)rangeFactor + offset;
+
+    if (inputs.at(hit.layer) >=
+        1022)  //the last address i.e. 1023 is reserved for the no-hit value, so interpolation between the 1022 and 1023 has no sense
+      inputs.at(hit.layer) = 1022;
+
+    if (print || inputs.at(hit.layer) < 0) {
+      edm::LogImportant("OMTFReconstruction")  //<<"rawData "<<hex<<setw(16)<<hit.rawData
+          << " layer " << dec << int(hit.layer);
+      edm::LogImportant("OMTFReconstruction")
+          << " phiDist " << hit.phiDist << " inputVal " << inputs.at(hit.layer) << " hit.z " << int(hit.z) << " valid "
+          << ((short)hit.valid) << " quality " << (short)hit.quality << " omtfRefLayer " << omtfRefLayer;
+      if (inputs.at(hit.layer) < 0)
+        edm::LogImportant("OMTFReconstruction") << " event->inputs.at(hit.layer) < 0 !!!!!!!!!!!!!!!!!" << endl;
+      edm::LogImportant("OMTFReconstruction") << endl;
+    }
+
+    if (inputs[hit.layer] >= 1024) {  //TODO should be the size of the LUT of the first layer
+      edm::LogImportant("OMTFReconstruction") << " event->inputs[hit.layer] >= 1024 !!!!!!!!!!!!!!!!!" << endl;
+    }
+    return true;
+  }
```

Comment:
There are several number literals here such as 63, 1022, or 1024 that I am not sure I understand the origin of. Could these be made named constants to try and offer a little explanation?

Reply:
@kbunkow, can you take a look, please?

Reply:
PtAssignmentNNRegression.cc will have to be modified again, as we have a new version of the neural network. This comment will be addressed in this new version (it will be in the next PR).